Cython/Utility/ObjectHandling.c

```diff
@@ -2352,6 +2352,65 @@ static CYTHON_INLINE PyObject* __Pyx_PyObject_FastCallDict(PyObject *func, PyObj
     #endif
 }
 
+/////////////// PyObjectFastCallCopyKwds.proto ///////////
+
+// Like __Pyx_PyObject_FastCallDict, but copies the kwargs dict in cases where we can't use PyObject_VectorcallDict
+// (to make sure that kwargs is not changed by the function call).
+// PyObject_VectorcallDict itself always ends up copying the dictionary.
+static CYTHON_INLINE PyObject* __Pyx_PyObject_FastCallDict_CopyKwds(PyObject *func, PyObject **args, size_t nargs, PyObject *kwargs); /*proto*/
+
+/////////////// PyObjectFastCallCopyKwds ///////////
+//@requires: PyObjectFastCall
+
+static CYTHON_INLINE PyObject* __Pyx_PyObject_FastCallDict_CopyKwds(PyObject *func, PyObject **args, size_t nargs, PyObject *kwargs) {
+    PyObject *result;
+    #if PY_VERSION_HEX < 0x03090000 || CYTHON_COMPILING_IN_LIMITED_API
+    if (kwargs) {
+        kwargs = PyDict_Copy(kwargs);
+        if (!kwargs) return NULL;
+    }
+    #endif
+    result = __Pyx_PyObject_FastCallDict(func, args, nargs, kwargs);
+    #if PY_VERSION_HEX < 0x03090000 || CYTHON_COMPILING_IN_LIMITED_API
+    Py_XDECREF(kwargs);
+    #endif
+    return result;
+}
+
+/////////////// PyObjectVectorCallKwBuilder.proto ////////////////
+//@requires: PyObjectFastCall
```

Comment:
The intention is that this can also be used by utility code. So something like the limited API `ByteArray.from_int` or `code.update` can be done efficiently by vectorcall when supported